### DISCUSSION TITLE
Self-Hosted and changing listen address and port and start at boot

### DISCUSSION THREAD

**Discussion by NightHawkATL** [healthchecks/healthchecks#807] [Q&A]
I have looked through all of the issues that could match the above keywords and have not found a solution. I am looking to do the following:
1. change the listen address from 127.0.0.1 to 0.0.0.0
2. create a service for the server and notifications and start them at boot
3. load the server through HTTPS using NginxProxyManager to start testing and then run it through Cloudflare Tunnels to control access.

I have looked through all of settings.py and and even tried to locate a few other settings files and realize that the setting is probably in the database but don't want to break it. 

I would also like to have all of the options that the cloud-hosted version has with integrations shown and also loading the images for the badges.

 I would lie to have some assistance in where to find any settings for what is listed above please.

**Comment by cuu508** [comment]
Hello @NightHawkATL, Healthchecks is a web application, written using the Django web framework. To run it in production, you need few additional components:

* WSGI HTTP server (for example: uwsgi, gunicorn).
* Reverse proxy (for example: nginx, haproxy, caddy). This is where you might do TLS termination
* A process manager to start the web server on boot (for example: systemd, supervisord)

For the [hosted service](https://healthchecks.io), I use uwsgi as the WSGI server, nginx as the reverse proxy, and systemd to start services automatically.

**Reply by NightHawkATL** [reply]
Thanks. I know it’s talked about towards the end but it may be helpful to provide examples of how you do it, just to help others. Do you have an example of a systemd start service?

**Comment by cuu508** [comment]
I'll think about publishing sample systemd service files somewhere. For now, here are the service files I'm using. The webapp code is at `/app/` and I'm using the system's python3 (and no virtualenv) because healthchecks is the only python app running on the machine. 

Systemd service for running uwsgi:

```ini
[Unit]
Description=healthchecks web server
After=network-online.target
Requires=network-online.target

[Service]
Slice=machine.slice
Restart=always
RestartSec=20
StartLimitInterval=10
StartLimitBurst=5

User=www-data
Group=www-data

ExecStart=/usr/bin/uwsgi-core /app/uwsgi.ini

PrivateDevices=true
ProtectHome=true
ProtectSystem=full

[Install]
WantedBy=multi-user.target
```

`/app/uwsgi.ini` is:

```ini
[uwsgi]
strict
master
die-on-term
uwsgi-socket = /tmp/healthchecks.sock
harakiri = 28
post-buffering = 16192
processes = 4
auto-procname
enable-threads
threads = 1
chdir = /app
plugin = python3
module = hc.wsgi:application
thunder-lock
disable-write-exception
disable-logging
```

Systemd service for running sendalerts:

```ini
[Unit]
Description=sendalerts
After=network-online.target
Wants=network-online.target

[Service]
Slice=machine.slice
Restart=always
RestartSec=20
StartLimitInterval=10
StartLimitBurst=5

User=www-data
Group=www-data
SyslogIdentifier=sendalerts

ExecStart=/usr/bin/python3 -u /app/manage.py sendalerts --no-threads

PrivateDevices=true
ProtectHome=true
ProtectSystem=full

[Install]
WantedBy=multi-user.target
```

**Reply by NightHawkATL** [reply]
I gave up on the VM/LXC approach and just run in Docker so it always starts at boot. you can close this as answered for others to see.
